bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/fileformat/FileFormatResource.java

```diff
@@ -0,0 +1,441 @@
+
+    private final ItemRegistry itemRegistry;
+    private final MetadataRegistry metadataRegistry;
+    private final ItemChannelLinkRegistry itemChannelLinkRegistry;
+    private final ThingRegistry thingRegistry;
+    private final Inbox inbox;
+    private final ThingTypeRegistry thingTypeRegistry;
+    private final ConfigDescriptionRegistry configDescRegistry;
+    private final Map<String, ItemSyntaxGenerator> itemSyntaxGenerators = new ConcurrentHashMap<>();
+    private final Map<String, ThingSyntaxGenerator> thingSyntaxGenerators = new ConcurrentHashMap<>();
+
+    @Activate
+    public FileFormatResource(//
+            final @Reference ItemRegistry itemRegistry, //
+            final @Reference MetadataRegistry metadataRegistry,
+            final @Reference ItemChannelLinkRegistry itemChannelLinkRegistry,
+            final @Reference ThingRegistry thingRegistry, //
+            final @Reference Inbox inbox, //
+            final @Reference ThingTypeRegistry thingTypeRegistry, //
+            final @Reference ConfigDescriptionRegistry configDescRegistry) {
+        this.itemRegistry = itemRegistry;
+        this.metadataRegistry = metadataRegistry;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+        this.thingRegistry = thingRegistry;
+        this.inbox = inbox;
+        this.thingTypeRegistry = thingTypeRegistry;
+        this.configDescRegistry = configDescRegistry;
+    }
+
+    @Deactivate
+    void deactivate() {
+    }
+
+    @Reference(policy = ReferencePolicy.DYNAMIC, cardinality = ReferenceCardinality.MULTIPLE)
+    protected void addItemSyntaxGenerator(ItemSyntaxGenerator itemSyntaxGenerator) {
+        itemSyntaxGenerators.put(itemSyntaxGenerator.getGeneratorFormat(), itemSyntaxGenerator);
+    }
+
+    protected void removeItemSyntaxGenerator(ItemSyntaxGenerator itemSyntaxGenerator) {
+        itemSyntaxGenerators.remove(itemSyntaxGenerator.getGeneratorFormat());
+    }
+
+    @Reference(policy = ReferencePolicy.DYNAMIC, cardinality = ReferenceCardinality.MULTIPLE)
+    protected void addThingSyntaxGenerator(ThingSyntaxGenerator thingSyntaxGenerator) {
+        thingSyntaxGenerators.put(thingSyntaxGenerator.getGeneratorFormat(), thingSyntaxGenerator);
+    }
+
+    protected void removeThingSyntaxGenerator(ThingSyntaxGenerator thingSyntaxGenerator) {
+        thingSyntaxGenerators.remove(thingSyntaxGenerator.getGeneratorFormat());
+    }
+
+    @GET
+    @RolesAllowed({ Role.ADMIN })
+    @Path("/existing/items")
+    @Produces("text/vnd.openhab-dsl")
+    @Operation(operationId = "createFileFormatForAllItems", summary = "Create file format for all existing items in registry.", security = {
+            @SecurityRequirement(name = "oauth2", scopes = { "admin" }) }, responses = {
+                    @ApiResponse(responseCode = "200", description = "OK", content = @Content(mediaType = "text/vnd.openhab-dsl", schema = @Schema(implementation = String.class))),
```

Comment:
```suggestion
                    @ApiResponse(responseCode = "200", description = "OK", content = @Content(mediaType = "text/vnd.openhab-dsl", schema = @Schema(example = "DSL Example"))),
```

Replace `DSL Example` by the example of a DSL output. You will then get something documented. It will look very similar to JSON output automatic examples.
You don't need to specificy the implementation class, as that will just put `string` as an example.

Reply:
Done